docs/development/adding-a-package-manager.md

```diff
@@ -67,3 +77,16 @@ This is used when more than one package manager shares settings from a common la
 ### `supportsLockFileMaintenance` (optional)
 
 Set to true if this package manager needs to update lock files in addition to package files.
+
+### `updateArtifacts` (async, optional)
+
+This function is triggered if a dependency has been updated for a package file or `lockfileMaintenance` has been triggered.
+Mostly this function is used to directly or indirectly ( e.g. by executing external binaries ) update lock files.
+
+### `updateDependency` (optional)
+
+`updateDependency` is used if the manager can not be updated with the standard replacing mechanisms and a custom replacement has to be provided.
+
+### `updateLockedDependency` (optional)
+
+???
```

Comment:
@rarkins @viceice 
Not sure what the scope exactly is, but seems we are calling this if we are able to update exactly a specific locked dependency?

Reply:
i think this is directly manipulating the lockfile. 🤔

Reply:
In that case we should move the Terraform lockfile support to `updateLockedDependency`, but not sure if it is triggered for `lockfileMaintenance`.